core/src/test/scala/org/apache/spark/util/EventLoopSuite.scala

```diff
@@ -185,4 +185,22 @@ class EventLoopSuite extends FunSuite with Timeouts {
     }
     assert(false === eventLoop.isActive)
   }
+
+  test("EventLoop: stop in eventThread") {
+    val eventLoop = new EventLoop[Int]("test") {
+
+      override def onReceive(event: Int): Unit = {
+        stop()
+      }
+
+      override def onError(e: Throwable): Unit = {
+      }
+
+    }
+    eventLoop.start()
+    eventLoop.post(1)
+    eventually(timeout(5 seconds), interval(200 millis)) {
```

Comment:
can we reduce the check interval to like 5ms? Ideally unit tests should run in a very short amount of time. Having to wait 200ms is pretty long. Would be great to change rest of this file also.